app/controllers/dashboard/internship_offers_controller.rb

```diff
@@ -23,20 +23,20 @@ def create
         end
         on.failure do |failed_internship_offer|
           @internship_offer = failed_internship_offer || InternshipOffer.new
-          find_selectable_weeks
+          @available_weeks = Week.selectable_from_now_until_school_year
           render :new, status: :bad_request
         end
       end
     rescue ActionController::ParameterMissing
       @internship_offer = InternshipOffer.new
-      find_selectable_weeks
+      @available_weeks = Week.selectable_from_now_until_school_year
       render :new, status: :bad_request
     end
 
     def edit
       @internship_offer = InternshipOffer.find(params[:id])
       authorize! :update, @internship_offer
-      find_selectable_weeks
+      @available_weeks = Week.selectable_on_school_year
```

Comment:
available weeks on edit shows all school year weeks (otherwise we 'lose the info' when time pass')